### ISSUE TITLE
Upgrade to Sequoia 15.0

### ISSUE THREAD

**Issue by macuserguru** [kasbert/OS-X-SAT-SMART-Driver#90]
I have no problems after upgrade to 15.0
OS-X-SAT-SMART-Driver works well, too. Download from DriveDX homepage, signed & compiled.
https://binaryfruit.com/drivedx/usb-drive-support
Runs on an M1 Mac Mini.

**Comment by mrmaximas** [comment]
Yes, but some modern drives and enclosures does nor supported with driveDx compiled driver version ;(
ioreg -r -w 0 -c fi_dungeon_driver_IOSATDriver  | egrep 'Enclosure|PassThroughMode|Capable'|grep -v Identifiers
it's works on Sequoia
  |   "PassThroughMode" = "sat16"
  |   "SATSMARTCapable" = Yes
  |   "Enclosure Name" = "MEDION HDDrive-n-GO"
  |   "KnownEnclosure" = Yes
it's Unknown... mentioned here https://www.smartmontools.org/ticket/1841
  |   "PassThroughMode" = "sat12"
  |   "IOStorageFeatures" = {"Unmap"=Yes,"UnmapCapable"=Yes}
  |   "Enclosure Name" = "Unknown 0bda:9210"
  |   "SATSMARTCapable" = Yes
  |   "KnownEnclosure" = No
@kasbert hot to fix it?

**Comment by voghuser** [comment]
I have a very peculiar and probably rare problem: SATSMART driver on a Mac mini M2 Pro with 15.1 (skipped 15.0) prevents Scubapro/UWATEC dive computers G2 and G2TEK to show their flash disk. No such problem on an Intel iMac. 
This flash disk is supposed to work like an USB stick, MS-DOS (FAT32), contains all data of the dive computer and is needed among other things for updates of the computer. This worked nicely for many years (7 years with the G2), various MacOS versions and SATSMART used for real USB hard disks. I. e. SATSMART ist by no means necessary for the dive computer. But now with MacOS 15 (15.1) the flash disk does not show up any more on the mini M2 Pro. However, it does show up on an iMac 2017 (Intel) with SATSMART and 15.1. When I delete SATSMART on the mini the flash disk shows up without any problems, apart from the usual problems with USB sticks and 15.1 (e.g. you have to wait for two minutes or so until size and free space become different from 0 in a finder window and you my write on the stick).
Yes, I confirm I have no problem with SATSMART and DriveDX.
So for my USB disks and DriveDX I need SATSMART, but it seems to keep the flash disk of the dive computer outside. Any ideas how to solve this problem? (I can use the old iMac for the time being, but ...) 
Edit Nov 21, 2024: Now with MacOS 15.1.1 I tried again. And suddenly (after 10 minutes or so) the flash disk of the dive computer appeared on my desktop and even in disk utility which happened to be open. To reproduce this I removed the flash disk and connected again. But even after restart of the Mac and waiting more than 20 minutes the flash disk did not show up again. So at least difficult to reproduce the observation. At least SATSMART is not the culprit, but seems to foster the problem on a M2 Mac mini. - Perhaps I should explicitly mention that apart from the 15.1 to 15.1.1 update I removed SATSMART completely and reinstalled and the system information now locates SATSMART as /Library/StagedExtensions/Library/Extensions/SATSMARTDriver.kext/ , whereas I still have /Library/Extensions/SATSMARTDriver.kext and /Library/Extensions/SATSMARTLib.plugin as expected, but not in system information. Independent of SATSMART I observe that MacOS does not like USB sticks with FAT32 any more. I always have to wait for at least two minutes until they show up. ExFAT versions seem to be much faster. So there is probably a nice mixture of problems and presumably SATSMART plays only a minor role (if any).
Edit Nov 22,2024: Never ending story! If there is no SATSMART driver (and Lib.plugin) and DriveDx was not launched (this is now new) then there are no problems both with one of the nasty sticks and the flash disk of the dive computers. They show up almost immediately after  plugged in and show the free space in the finder window immediately after klicking on the stick. Behavior is then just like safe mode. With DriveDx launched (still no SATSMART installed) then it takes two minutes or so until free space different from 0 is displayed closing and opening the finder window of the stick to refresh the display). No idea what DriveDx should do without SATSMART. (And I suppose that even with SATSMART these sticks should not be expected to show up in DriveDx.)

**Comment by ferdnyc** [comment]
@voghuser I don't have any help to offer with your issue, but you should _definitely_ be posting it as a new issue of your own, not as a comment on @macuserguru 's report about successfully running DriveDX on macOS 15. There's almost no chance your issue will get any attention, where it's currently posted.

(And you should probably be posting updates as new, separate comments, not by editing them into your one bigmclargehuge comment. Because it's turned into a wall of text that's growing increasingly harder to read.)

Blank lines between paragraphs would help a lot.

**Comment by AlecKinnear** [comment]
For those with Apple Silicon, I've just successfully installed WD Drive Utilities which includes a basic SMART check via USB for WD drive enclosures (MyBook in this case), as well as both a quick test and an extended test. This is basically the DriveDX features without the granularity (one doesn't see On hours for instance or defective sectors).

I'm a big DriveDX fan but I've had too much trouble trying to get SatSmart Driver to work with DriveDX on my Apple Mx Macs. I've basically given up on DriveDX for USB drives. With NVME drives, I rotate them into a Thunderbolt housing for occasional checks but that doesn't help with my backup hard drives.

**Comment by macuserguru** [comment]
“WD Drive Utilities“, but I use Seagate/LaCie, Crucial Micron (USB SSD) drives in addition to WD drives.
Look at [https://github.com/smartmontools/smartmontools/issues/373](url)

For Crucial Micron SSD - in  DriveDX I could not see the Crucial X10 SSD drive, in 
SMARTReporter / Version:  3.2.0, Build: 2576, CoreCode
and
SMART Utility, Version 3.2.7 (3C142), ©2022 Volitans Software
yet.

I installed smartmontools with Homebrew `smartctl -a ....“drive“` works with X10 SSD too.

My system - Apple Silicon M1 MacMini, Sequoia 15.6
